usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh

```diff
@@ -0,0 +1,27 @@
+is_true $EFI_STUB || return 0
+
+# Despite user made his choice, check if kernel can really boot with EFISTUB.
+grep -q -i "EFI stub" $KERNEL_FILE || Error "$KERNEL_FILE is not compiled with EFISTUB support"
+Log "EFI_STUB: Using kernel $KERNEL_FILE with EFISTUB support"
+
+Log "EFI_STUB: Checking if Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
+esp_mountpoint=$( df -P "${KERNEL_FILE}" | tail -1 | awk '{print $6}' )
+uefi_fs_type=$(df -T "$esp_mountpoint" | tail -n +2 | head -n 1 | awk '{print $2}')
+
+if [[ "$uefi_fs_type" != "vfat" ]]; then
+    Error "EFI_STUB: Kernel file: $KERNEL_FILE is not hosted on vfat filesystem"
+else
+    Log "EFI_STUB: Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
+fi
+
```

Comment:
Ony a side note:
You could - if you like - simplify it e.g. as
```
if ERROR_CONDITION ; then
    Error "..."
fi
Log "..."
```
or even
```
ERROR_CONDITION && Error "..."
Log "..."
```

Reply:
Something like:
```
[[ "$uefi_fs_type" != "vfat" ]] && Error "EFI_STUB: Kernel file: $KERNEL_FILE is not hosted on vfat filesystem"
Log "EFI_STUB: Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
```
?

Reply:
Yes - but only if you like it - I see such kind of code e.g. in
usr/share/rear/layout/save/GNU/Linux/190_opaldisk_layout.sh